src/graph.jl

```diff
@@ -0,0 +1,74 @@
+using JSON
+using SparseArrays
+using LightGraphs
+abstract type AbstractGraph end
+
+struct Graph<:AbstractGraph
+    num_nodes::Int
+    num_edges::Int
+
+    # an adjacency matrix where each entry is the edge id that connects
+    # the two nodes.
+    adj_matrix::SparseMatrixCSC{Int, Int}
+
+    # source and destination of the edge. the index of these arrays are the
+    # edge ids.
+    edge_src::Array{Int, 1}
+    edge_dst::Array{Int, 1}
+
+    # an Array of Arrays where the neighbors of node `i` are found in an array
+    # at index `i`
+    node_neighbors::Array{Array{Int64,1},1}
+
+    # the base SimpleGraph, if we need it
+    simple_graph::SimpleGraph
+end
+
+function Graph(raw_graph::Dict{String, Any},
+               nodes_str::AbstractString = "nodes",
+               adjacency_str::AbstractString = "adjacency",
+               edge_id_str::AbstractString = "id")::Graph
+
+    """ Builds the base Graph object. This is the underlying network of our
+        districts, and should never need to be changed.
+
+        Arguments:
+            filepath: file path to the .json file that contains the graph. This
+                      file is expected to be generated by the `Graph.to_json()`
+                      function http://tiny.cc/7ya2jz of the Python implementation
```

Comment:
Not sure about the `tiny.cc` thing here, as it obscures that it's a reference to the GerryChain documentation... though there's not really a good solution for handling long URLs in docstrings. :\

Reply:
leave a citation for the url instead of the tiny cc

Reply:
The citation thing turns out to still be rather ugly... but I am fine leaving the long url here.